src/main/java/sirius/biz/storage/layer2/BasicBlobStorageSpace.java

```diff
@@ -1410,6 +1410,16 @@ protected String lookupPhysicalKey(String blobKey, @Nonnull String variantName,
         }
     }
 
+    /**
+     * Ensures that the desired variant exists.
+     *
+     * @param blob        the blob for which the variant should exist
+     * @param variantName the name of the desired variant
+     */
+    public void ensureVariantExists(B blob, String variantName) {
```

Comment:
Why do we need the whole method around, instead of just making findOrCreateVariant public, which even delivers the variant created or obtained?

Reply:
I was pair programming with @andyHa and he suggested that we build the method like this. The idea was to shield as much as possible from a caller and only provide a simple mechanism to trigger the variant creating if necessary.